### ISSUE TITLE
'EphemeralMetrics' object has no attribute 'set_default_dimensions'

### ISSUE THREAD

**Issue by tibbe** [awslabs/aws-lambda-powertools-python#2676]
### Expected Behaviour

`log_metrics()` with `default_dimensions` should also work for `EphemeralMetrics`.

### Current Behaviour

Since `EphemeralMetrics` has no `set_default_dimensions` method it fails with the error in the issue title.

### Code snippet

```python
@my_metrics.log_metrics(
    default_dimensions={'Stage': 'prod'}
):
def handler(event, context):
   pass:
```

### Possible Solution

`EphemeralMetrics` should have most of the methods that `Metrics` has (in particular those related to default dimensions), as these method are orthogonal to the non-singletoness of `EphemeralMetrics`.

### Steps to Reproduce

Run a Lambda with the code snippet above.

### Powertools for AWS Lambda (Python) version

latest

### AWS Lambda function runtime

3.10

### Packaging format used

PyPi

**Comment by heitorlessa** [comment]
hey @tibbe great to hear from you again - [we intentionally did that ](https://docs.powertools.aws.dev/lambda/python/latest/core/metrics/#metrics-isolation).

If you have an use case to [persist dimensions across invocations](https://docs.powertools.aws.dev/lambda/python/latest/core/metrics/#adding-default-dimensions) within ephemeral metrics, happy to convert this to a feature request -- if you have the bandwidth to make a PR we can prioritize it for the next release :) 


<img width="1175" alt="image" src="https://github.com/aws-powertools/powertools-lambda-python/assets/3340292/264f3a9e-796b-44ca-827a-44150dfca587">

**Comment by heitorlessa** [comment]
Asked on Discord to bring more visibility: https://discord.com/channels/1006478942305263677/1006527338621710376/1127918302686097438

In hindsight, I'm starting to think this makes sense. We'd need to signal in the docs that `EphemeralMetrics` instances will not share a default dimension since it's completely isolated.

I'll wait for a signal from you or another customer before proceeding.

Thanks!

**Comment by heitorlessa** [comment]
Confirmed use case on Discord, will get this ready for the next release.

---

from OP

I have a Lambda that consumes a DynamoDB stream and receives a batch of items from DynamoDB in each invocation. I have three metrics that have a dimension that is the same across the batch (the "Stage" e.g. "prod" or "development") and a dimension that varies across the batch (the "ItemType" e.g. "user" or "chat"). The former dimension would be convenient to set as a default dimension. I could of course just set it directly after the invocation starts, which is what I did as a workaround.

The dimension that varies across the batch is a bit trickier, as I can't re-set a dimension (all metrics would simply get the value of the last call to set_dimension). I ended up using a dict of EphemeralMetrics to handle this.

At the end of the day I'm just trying to use Lambda metrics in the way that's most efficient. The ideal API is actually just to emit the dimensions with the metric (i.e. just as a SingleMetric or really just a call to a hypothetical emit_metric(*dimensions, name, unit)). What we're doing here is in my mind just a workaround for what the Lambda logging schema supports efficiently.
